test/trento/application/integration/discovery/protocol/enrich_request_host_deregistration_test.exs

```diff
@@ -0,0 +1,63 @@
+defmodule Trento.EnrichRequestHostDeregistrationTest do
+  use ExUnit.Case
+  use Trento.DataCase
+
+  import Trento.Factory
+
+  alias Trento.Domain.Commands.RequestHostDeregistration
+  alias Trento.Support.Middleware.Enrichable
+
+  describe "enrich RequestHostDeregistration" do
+    test "should return the original command if a host with critical health is requested to be deregistered" do
```

Comment:
please, specify more the condition, critical health could be misleading, we don't pass health critical in these tests, we specify a "old hearbeat", rephrase the tests with this in mind